src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -62,54 +62,74 @@ public function validate(mixed $entity, Constraint $constraint)
             throw new ConstraintDefinitionException('At least one field has to be specified.');
         }
 
-        if (null === $entity) {
+        if (null === $value) {
             return;
         }
 
-        if (!\is_object($entity)) {
-            throw new UnexpectedValueException($entity, 'object');
+        if (!\is_object($value)) {
+            throw new UnexpectedValueException($value, 'object');
         }
 
+        $entityClass = $value::class;
+
         if ($constraint->em) {
             $em = $this->registry->getManager($constraint->em);
 
             if (!$em) {
                 throw new ConstraintDefinitionException(sprintf('Object manager "%s" does not exist.', $constraint->em));
             }
         } else {
-            $em = $this->registry->getManagerForClass($entity::class);
+            $em = $this->registry->getManagerForClass($constraint->entityClass ?? $entityClass);
```

Comment:
Is this, effectively, a bug fix? Previously, even if `$constraint->entityClass` was set, we used `$entity::class` to get the entity manager. Was that on purpose or does this fix a bug?

Reply:
Also:

```suggestion
            $em = $this->registry->getManagerForClass($constraint->entityClass ?? $value::class;);
```

This is more clear to me. Then we can remove the `$entityClass = ` from line 71 above. Or, if it's still needed for some other reason, move it lower to where it's needed.

Reply:
About the first comment.
Currently, the [entityClass](https://symfony.com/doc/current/reference/constraints/UniqueEntity.html#entityclass) argument is used to solve inheritance problems. When validating any class (like DTO), it is required because the passed object is not an entity.
I had to remember what it currently does. It would probably be useful to make a method with a meaningful name e.g. `getEntityClass`.

Reply:
> Currently, the [entityClass](https://symfony.com/doc/current/reference/constraints/UniqueEntity.html#entityclass) argument is used to solve inheritance problems

Yea, I know - this fact is what makes this PR so delicate / complex. Do your best to make the code clear